spikeinterface/sortingcomponents/dimensionality_reduction.py

```diff
@@ -0,0 +1,96 @@
+from sklearn.decomposition import IncrementalPCA
+from sklearn.preprocessing import FunctionTransformer
+from sklearn.pipeline import Pipeline
+
+from spikeinterface.sortingcomponents.peak_pipeline import run_peak_pipeline, PeakPipelineStep
+from spikeinterface.sortingcomponents.peak_detection import detect_peaks
+
+from joblib import dump, load
+
+class WaveformExtractorStep(PeakPipelineStep):
+    need_waveforms = True
+
+    def __init__(self, recording, ms_before=1., ms_after=1.,  peak_sign='neg', all_channels=True):
+        PeakPipelineStep.__init__(self, recording, ms_before=ms_before, ms_after=ms_after)
+        self.all_channels = all_channels
+        self.peak_sign = peak_sign
+        self._kwargs.update(dict(all_channels=all_channels, peak_sign=peak_sign))
+        self._dtype = recording.get_dtype()
+
+    def get_dtype(self):
+        return self._dtype
+
+    
+    def compute_buffer(self, traces, peaks, waveforms):
+        
+        return waveforms
+
+def stack_channel_with_spikes(waveform):
+    n_spikes, n_samples, n_channels = waveform.shape 
+
+    return waveform.transpose(0, 2, 1).reshape(n_spikes * n_channels, n_samples)
+
+def separate_channel_from_spikes(temporal_staggered_waveform, n_components, n_channels):
+    n_spikes = int(temporal_staggered_waveform.shape[0] / n_channels)
+    reshaped = temporal_staggered_waveform.reshape(n_spikes, n_channels, n_components)
+    return reshaped.transpose(0, 2, 1)
```

Comment:
not needed (we can use the `postprocessing` module)

Reply:
All right, I wanted to use standard scikit-learn machinery instead of the ad-hoc concatenation that you have for this method there. I will change it.